Moved vendor specific HCI to HAL

mtpr-ot opened the pull request:
Moved vendor HCI code to new HAL folder to allow out-of-tree
vendor specific HCI implementations.

cvinayak commented on include/bluetooth/hci_vs.h:
```diff
@@ -25,7 +26,8 @@ extern "C" {
 #define BT_HCI_VS_FW_VAR_RESCUE_IMG             0x0004
-#define BT_HCI_OP_VS_READ_VERSION_INFO		BT_OP(BT_OGF_VS, 0x0001)
+
+#define BT_HCI_OP_VS_READ_VERSION_INFO			BT_OP(BT_OGF_VS, 0x0001)
```
Please remove tabs (use spaces to align), and align with other BT_OP

cvinayak commented on subsys/bluetooth/controller/CMakeLists.txt:
```diff
@@ -60,6 +67,14 @@ zephyr_library_include_directories(
   )
 
+if(CONFIG_SOC_COMPATIBLE_NRF)
+if(CONFIG_BT_HCI_VS)
```
indent the second if clause block

cvinayak reviewed (changes requested):
Minor comments, otherwise LGTM.

mtpr-ot committed:
bluetooth: controller: Moved vendor specific HCI to HAL

Moved vendor HCI code to new HAL folder to allow out-of-tree
vendor specific HCI implementations.

Signed-off-by: Morten Priess <mtpr@oticon.com>

aescolar commented:
@mtpr-ot : Should this be closed, being replaced by #14143 ?

mtpr-ot commented:
> @mtpr-ot : Should this be closed, being replaced by #14143 ?
Yes, you're right. Will close now. Thanks.
